Source/Router.swift

```diff
@@ -210,14 +213,31 @@ public final class Router {
 
         let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(latency * Double(NSEC_PER_SEC)))
         dispatch_after(delayTime, dispatch_get_main_queue()) {
-            didFinishLoading(server)
+            [weak self] in
+            // before reporting "finished", check if request has been canceled in the meantime
+            guard let strongSelf = self else {
+                return
+            }
+            if let serverIndex = strongSelf.canceledRequests.indexOf(server) {
+                // remove server from the list of "canceled requests" and DO NOT send notification(s)
+                strongSelf.canceledRequests.removeAtIndex(serverIndex)
+            }
+            else {
+                didFinishLoading(server)
+            }
+        }
+    }
+
+    func stopLoading(server: NSURLProtocol) {
+        if canceledRequests.contains(server) == false {
```

Comment:
Actually `KakapoServer` instances are automatically created by the operating system for those request for which the `KakapoServer`'s `canInitWithRequest(request: NSURLRequest)` method returned `true`.

This makes it really hard to actually create a one-to-one relationship between `Router` and `KakapoServer`.

I solved this issue by remembering the `NSURL`s which should be stopped.
See c10b110fa7b1f30d1a5c9ad6a97e81917c99958f